src/Riok.Mapperly/Descriptors/MappingBuilders/StringToEnumMappingBuilder.cs

```diff
@@ -75,4 +95,57 @@ private static EnumFallbackValueMapping BuildFallbackParseMapping(MappingBuilder
             new EnumFromStringParseMapping(ctx.Source, ctx.Target, genericEnumParseMethodSupported, ctx.Configuration.Enum.IgnoreCase)
         );
     }
+
+    private static IReadOnlyDictionary<IFieldSymbol, AttributeValue> BuildExplicitValueMappings(MappingBuilderContext ctx)
+    {
+        var explicitMappings = new Dictionary<AttributeValue, IFieldSymbol>(new AttributeValueByConstantValueEqualityComparer());
+        foreach (var (source, target) in ctx.Configuration.Enum.ExplicitMappings)
+        {
+            var targetField = target.ConstantValue.GetEnumField()!;
+
+            if (!SymbolEqualityComparer.Default.Equals(target.ConstantValue.Type, ctx.Target))
+            {
+                ctx.ReportDiagnostic(
+                    DiagnosticDescriptors.TargetEnumValueDoesNotMatchTargetEnumType,
+                    targetField,
+                    target.ConstantValue.Value ?? 0,
+                    target.ConstantValue.Type?.ToDisplayString() ?? "unknown",
+                    ctx.Target
+                );
+                continue;
+            }
+
+            if (explicitMappings.ContainsKey(source))
```

Comment:
Instead of the custom comparer + reversing the resulting dictionary it may probably be easier / faster to keep a separate hashset of the processed source values for duplicate checking (e.g. `if (!sources.Add(source.ConstantValue.Value))`)

Reply:
Yes, agreed. I have made this change.